### PR TITLE
docs: clarify func-style

### PR DESCRIPTION
**Pull request by camsteffen**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

* Say "function expressions assigned to variables" in some cases for clarity
* Prefer `const` over `var` in examples
* Some tidying/rewording
* Add a note saying the lint does not apply to all functions

#### Is there anything you'd like reviewers to focus on?
Fixes: #18474

### CODE REVIEW

File: docs/src/rules/func-style.md [hunk 1]
```diff
@@ -5,41 +5,41 @@ further_reading:
...
 ```js
+// function declaration
 function doSomething() {
     // ...
 }
-```
 
-Equivalent function expressions begin with the `var` keyword, followed by a name and then the function itself, such as:
+// arrow function expression assigned to a variable
+const doSometing = () => {
```
Comment (nzakas) [inline docs/src/rules/func-style.md:17]: Let's use unique IDs to avoid syntax errors.

```suggestion
const doSomethingElse = () => {
```

**Comment by Tanujkanti4441** [comment]
Hi @camsteffen, some requested changes are yet to be implemented, can you implement the rest also?

**Comment by camsteffen** [comment]
*facepalm* I'm sorry, I had the changes locally but didn't actually push them up.

**Review by snitin315 — APPROVED** [review]
LGTM. I will leave it open for @nzakas

File: docs/src/rules/func-style.md [hunk 2]
```diff
@@ -50,7 +50,9 @@ Due to these different behaviors, it is common to have guidelines as to which st
 
 ## Rule Details
 
-This rule enforces a particular type of `function` style throughout a JavaScript file, either declarations or expressions. You can specify which you prefer in the configuration.
+This rule enforces a particular type of function style, either `function` declarations or expressions assigned to variables. You can specify which you prefer in the configuration.
+
+Note: This lint does not apply to *all* functions. For example, a callback function passed as an argument to another function is not considered by this lint.
```
Comment (mdjermanovic) [inline docs/src/rules/func-style.md:55]:
```suggestion
Note: This rule does not apply to *all* functions. For example, a callback function passed as an argument to another function is not considered by this rule.
```

**Review by mdjermanovic — CHANGES_REQUESTED** [review]
We don't use the term "lint" for rules.

**Review by nzakas — APPROVED** [review]
LGTM. I applied @mdjermanovic's suggestion, so this is good to go. Thanks!

**Pull request merged**